docs/utilities/config_annotations.md

```diff
@@ -0,0 +1,234 @@
+Configuration Annotations
+=========================
+
+The 1.12.x version of Minecraft Forge introduced novel mod configuration system based on annotations. This new system relies on Forge ASM as to automatically discover mod configuration classes and their respective configurable fields as to make working with configurations as easy as possible. Ideally this new system will encourage the standard implementation of mod configurations as to afford end-users with maximally configurable modded Minecraft experiences.
+
+
+Basics
+-------------------------
+
+The annotation-based configuration system revolves around the `@Config` annotation, found in the `net.minecraftforge.common.config` package of Minecraft Forge. The `@Config` annotation is intended to be applied to any class that contains user-configurable fields, requiring only the id of the mod that owns the configuration for basic usage. Following the application of the `@Config` annotation to a configuration class, Forge will then resolve and manage any `public static` fields in the class as configuration elements.
+
+!!! important
+
+    There is currently no way to have Forge ignore a `public static` field in a `@Config` class, so make sure not to place any non-user configurable `public static`
+
+Consequently, the simplest possible usage of the annotation-based configuration system would be as follows.
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  public static boolean myBooleanSetting = false;
+
+}
```

Comment:
Since this is just a brief example, the two empty lines are probably unnecessary and take up room.